bundles/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/internal/ErrorHandler.java

```diff
@@ -0,0 +1,151 @@
+
+    private void offLine(final int status, Object device) {
+        String err = Errors.getErrorMessage(status);
+
+        this.logger.warn("Warning: " + status + (err == null ? " " : err));
+
+        if (UDClientStatus.isBusy()) {
+            return;
+        }
+        if (status == 1050) {
+            this.socket_open_failed_count++;
+        }
+        if (this.socket_open_failed_count < 3) {
+            return;
+        }
+        this.socket_open_failed_count = 0;
+
+        if (device == null) {
+            device = UDControlPoint.firstDevice;
+        }
+        if (device != null && device instanceof UDProxyDevice) {
+            UDControlPoint.getInstance().offLine((UDProxyDevice) device);
+        }
+
+    }
+
+    private void humanInterventionRequired(final int status, final String msg) {
+        String err = Errors.getErrorMessage(status);
+        this.logger.error("Human Intervention Required: " + status + (err == null ? " " : err));
+        if (msg != null) {
+            this.logger.error(msg);
+        }
+    }
+
+    private void warning(final int status, final String msg) {
+        String err = Errors.getErrorMessage(status);
+
+        this.logger.warn("Warning: " + status + (err == null ? " " : err));
+        if (msg != null) {
+            this.logger.warn(msg);
+        }
+    }
+
+    private void fatalError(final int status, final String msg) {
+        String err = Errors.getErrorMessage(status);
+        this.logger.error("Fatal Error: " + status + (err == null ? " " : err));
+        if (msg != null) {
+            this.logger.error(msg);
+        }
+    }
+
+    /**
+     * This method is invoked when ISY encounters an error
+     * 
+     * @param status
+     *            - the error code
+     * @param msg
+     *            - any generated messages by <code>ISYClient</code>
+     * @param device
+     *            - the <code>UDProxyDevice</code> initiating this event
+     * @return - whether or not the client attempt displaying the error on a UI.
+     *         if a <code>GUIErrorHandler</code> is installed, that object is
+     *         invoked (showError)
+     */
+
+    @Override
+    public boolean errorOccured(final int status, final String msg, final Object device) {
+
+        switch (status) {
+            case -1:
+            case 803/* discovering nodes:retry */:
+            case 902/* Node Is in Error; Check connections! */:
+                warning(status, null);
+                break;
+
+            case 801/* maximum secure sessions */:
+            case 805/* internal error: reboot */:
+            case 815/* Maximum Subscribers Reached */:
+            case 903/* System Not Initialized; Restart! */:
+            case 905/* Subscription Failed; The device might need reboot! */:
+            case 1020/* Couldn't create the event handler socket */:
+                fatalError(status, "Exit the applciation; might have to reboot ISY");
```

Comment:
spelling fixed -- I don't know about the error handling, perhaps @sytone should comment, I am not that familiar with the algorithms used here.